app/src/common/shared/com/igalia/wolvic/VRBrowserActivity.java

```diff
@@ -683,26 +681,49 @@ void loadFromIntent(final Intent intent) {
             }
         }
 
-        Uri uri = intent.getData();
-
         boolean openInWindow = false;
         boolean openInBackground = false;
+        boolean openInKioskMode = false;
+
+        Uri dataUri = intent.getData();
+        Uri targetUri = null;
+        Bundle extras;
+
+        if (dataUri != null && dataUri.getScheme().equals("wolvic") && dataUri.getHost().equals("com.igalia.wolvic")) {
+            Log.d(LOGTAG, "Parsing custom URI from intent: " + dataUri);
+
+            extras = new Bundle();
+            Set<String> keys = dataUri.getQueryParameterNames();
+            for (String key : keys) {
+                String queryParameter = dataUri.getQueryParameter(key);
+                if (queryParameter == null)
+                    continue;
+
+                if (key.equalsIgnoreCase("url") || key.equalsIgnoreCase("homepage"))
```

Comment:
We're treating `url` and `homepage` the same, so I think we can get rid of the `homepage` one as it does add nothing.

Reply:
The `"homepage"` parameter changes the home page that is stored in the Settings.

I have removed it, it is strange to do this without user input. Firefox in the desktop doesn't have this option and Chrome only uses it for testing.